src/main/java/io/goobox/sync/storj/CheckStateTask.java

```diff
@@ -70,10 +69,12 @@ public void onFilesReceived(String bucketId, File[] files) {
                 DB.commit();
 
                 if (tasks.isEmpty()) {
+                    if (isSync == true) {
```

Comment:
> he added upload task was taken too quickly by the TaskExecutor, so it started before the CheckStateTask finished. This way the 'idle' event was sent too early - while the file was still uploading. It's an old issue, unrelated to your work

> Isn't it better to move this check inside the updateIdleEvent() method? Just like you have for the updateSyncEvent().

Yes, when uploading large files (during tests run locally here), the uploading still happening and `tasks` is not empty. So there is a check to `idle` if it is false, then only send one idle event out.

For `setSynchronizing`, I did it in the method because this method is called in multiple other methods, the logic is not to keep repeating in other methods but in this method.

For `setIdle`, if logic was chosen in the method `onFilesReceived` because I was thinking to place all logic in this method whilst `setIdle` just leave it as minimal. But for this situation is trivial and if you think the if logic in `setIdle` is more comprehendable, then okay.